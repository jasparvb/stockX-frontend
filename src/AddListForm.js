import React, { useState } from "react";
import { useDispatch } from 'react-redux';
import { addNewListAPI } from './actions/lists';

//Form to create a new list

function AddListForm() {
  const dispatch = useDispatch();

  const INITIAL_STATE = { 
    title: ""
  };

  const [text, setText] = useState(INITIAL_STATE);

  function handleSubmit(evt) {
    evt.preventDefault();
    dispatch(addNewListAPI(text));
    setText(INITIAL_STATE);
  };

  /** Update local state w/curr state of input elem */

  const handleChange = evt => {
    const { value } = evt.target;
    setText({title: value});
  };

  return (
    <div className="AddListForm">
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <input 
            name="title" 
            type="text"
            className="form-control" 
            placeholder="New list title"
            value={text.title} 
            onChange={handleChange}
          />
        </div>
        <button type="submit" className="btn btn-primary">Add</button>
      </form>
    </div>
  );
}

export default AddListForm;