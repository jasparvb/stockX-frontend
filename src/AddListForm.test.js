import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import rootReducer from "./reducers/root";
import { BrowserRouter } from "react-router-dom";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';
import AddListForm from './AddListForm';

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
);

it("renders without crashing", function() {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <AddListForm />
      </BrowserRouter>
    </Provider>);
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddListForm />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});