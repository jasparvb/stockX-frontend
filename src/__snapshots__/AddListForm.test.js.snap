// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddListForm"
  >
    <form>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="title"
          placeholder="New list title"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
